assets/src/edit-story/components/canvas/navLayer.js

```diff
@@ -17,26 +17,53 @@
 /**
  * External dependencies
  */
-import { memo } from 'react';
+import { memo, useRef } from 'react';
+import { useFeature } from 'flagged';
 
 /**
  * Internal dependencies
  */
+import { ContextMenu } from '../../../design-system';
+import { useQuickActions } from '../../app/highlights';
+import DirectionAware from '../directionAware';
 import Header from '../header';
 import Carousel from '../carousel';
-import { Layer, HeadArea, CarouselArea, Z_INDEX } from './layout';
+import {
+  Layer,
+  HeadArea,
+  CarouselArea,
+  Z_INDEX,
+  QuickActionsArea,
+} from './layout';
 
 function NavLayer() {
+  const carouselAreaRef = useRef(null);
+  const headAreaRef = useRef(null);
+  const enableQuickActionMenu = useFeature('enableQuickActionMenus');
+  const quickActions = useQuickActions();
+
   return (
     <Layer
       pointerEvents="none"
       zIndex={Z_INDEX.NAV}
       onMouseDown={(evt) => evt.stopPropagation()}
     >
-      <HeadArea pointerEvents="initial">
+      <HeadArea pointerEvents="initial" ref={headAreaRef}>
```

Comment:
🤔 what are we using the refs for?

Reply:
Oh good call outs. Let me clean up a bit. That was all for a prior approach where the tab functionality was all controlled

Reply:
eventually figured If I'm recreating native tab functionality, I should just use native tab functionality lol